src/error.rs

```diff
@@ -7,15 +7,19 @@ use thiserror::Error;
 use crate::report::DerivationTree;
 use crate::solver::DependencyProvider;
 
+/// There is no solution for this set of dependencies.
+pub type NoSolutionError<DP> = DerivationTree<
+    <DP as DependencyProvider>::P,
+    <DP as DependencyProvider>::VS,
+    <DP as DependencyProvider>::M,
+>;
```

Comment:
I'm not really sure about this; should we rename `DerivationTree` to `NoSolutionError`, or should this be a newtype?

Reply:
Derivation Tree is the technical term used in the pubgrub algorithm description so it’s good that it exists in the code base. An alias like this is fine to me.